opentracing/tracecontext.go

```diff
@@ -14,55 +14,47 @@ import (
 // TraceContexts are sufficient to propagate the, well, *context* of a
 // particular trace between processes.
 //
-// TraceContext also support a simple string map of "trace tags". These trace
-// tags are special in that they are propagated *in-band*, presumably alongside
-// application data. See the documentation for SetTraceTag() for more details
-// and some important caveats.
+// TraceContext also support a simple string map of "trace attributes". These
+// trace attributes are special in that they are propagated *in-band*,
+// presumably alongside application data. See the documentation for
+// SetTraceAttribute() for more details and some important caveats.
 type TraceContext interface {
-	// NewChild creates a child context for this TraceContext, and returns both
-	// that child's own TraceContext as well as any Tags that should be added
-	// to the child's Span.
+	// SetTraceAttribute sets a tag on this TraceContext that also propagates
+	// to future TraceContext children per TraceContext.NewChild.
 	//
-	// The returned TraceContext type must be the same as the type of the
-	// TraceContext implementation itself.
-	NewChild() (childCtx TraceContext, childSpanTags Tags)
-
-	// SetTraceTag sets a tag on this TraceContext that also propagates to
-	// future TraceContext children per TraceContext.NewChild.
-	//
-	// SetTraceTag() enables powerful functionality given a full-stack
+	// SetTraceAttribute() enables powerful functionality given a full-stack
 	// opentracing integration (e.g., arbitrary application data from a mobile
 	// app can make it, transparently, all the way into the depths of a storage
 	// system), and with it some powerful costs: use this feature with care.
 	//
-	// IMPORTANT NOTE #1: SetTraceTag() will only propagate trace tags to
-	// *future* children of the TraceContext (see NewChild()) and/or the
-	// Span that references it.
+	// IMPORTANT NOTE #1: SetTraceAttribute() will only propagate trace
```

Comment:
ic.. this is the confusing part. PropagationAttribute would make more sense in my brain, as it only affects downstream, not the enclosing trace.